src/components/molecules/ChatsList/ChatsList.tsx

```diff
@@ -0,0 +1,190 @@
+  Fragment,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from "react";
+import { isEmpty } from "lodash";
+
+import { DEFAULT_PARTY_NAME, VENUE_CHAT_AGE_DAYS } from "settings";
+
+import { User } from "types/User";
+
+import { getDaysAgoInSeconds, roundToNearestHour } from "utils/time";
+import { WithId } from "utils/id";
+import { chatUsersSelector, privateChatsSelector } from "utils/selectors";
+
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+
+import {
+  ChatContext,
+  PrivateChatMessage,
+} from "components/context/ChatContext";
+import UserProfilePicture from "components/molecules/UserProfilePicture";
+import ChatBox from "components/molecules/Chatbox";
+import { setPrivateChatMessageIsRead } from "components/organisms/PrivateChatModal/helpers";
+
+import "./ChatsList.scss";
+
+interface LastMessageByUser {
+  [userId: string]: PrivateChatMessage;
+}
+
+const noopHandler = () => {};
+
+const ChatsList: React.FunctionComponent = () => {
+  const { user } = useUser();
+  const privateChats = useSelector(privateChatsSelector);
+  const chatUsers = useSelector(chatUsersSelector);
+
+  const [selectedUser, setSelectedUser] = useState<WithId<User>>();
+
+  const lastMessageByUserReducer = useCallback(
+    (agg, item) => {
+      let lastMessageTimeStamp;
+      let discussionPartner;
+      if (item.from === user?.uid) {
+        discussionPartner = item.to;
+      } else {
+        discussionPartner = item.from;
+      }
+      try {
+        lastMessageTimeStamp = agg[discussionPartner].ts_utc;
+        if (lastMessageTimeStamp < item.ts_utc) {
+          agg[discussionPartner] = item;
+        }
+      } catch {
+        agg[discussionPartner] = item;
+      }
+      return agg;
+    },
+    [user]
+  );
+
+  const discussionPartnerWithLastMessageExchanged = useMemo(() => {
+    if (!privateChats) return {};
+    return privateChats.reduce<LastMessageByUser>(lastMessageByUserReducer, {});
+  }, [lastMessageByUserReducer, privateChats]);
+
+  const onClickOnSender = useCallback(
+    (sender: WithId<User>) => {
+      const chatsToUpdate = privateChats.filter(
+        (chat) => !chat.isRead && chat.from === sender.id
+      );
+      chatsToUpdate.map(
+        (chat) => user && setPrivateChatMessageIsRead(user.uid, chat.id)
+      );
+      setSelectedUser(sender);
+    },
+    [privateChats, user]
+  );
+
+  const DAYS_AGO = getDaysAgoInSeconds(VENUE_CHAT_AGE_DAYS);
+  const HIDE_BEFORE = roundToNearestHour(DAYS_AGO);
+
+  const chatsToDisplay = useMemo(
+    () =>
+      privateChats &&
+      privateChats
+        .filter(
+          (message) =>
+            message.deleted !== true &&
+            message.type === "private" &&
+            (message.to === selectedUser?.id ||
+              message.from === selectedUser?.id) &&
+            message.ts_utc.seconds > HIDE_BEFORE
+        )
+        .sort((a, b) => b.ts_utc.valueOf().localeCompare(a.ts_utc.valueOf())),
```

Comment:
[NB] This would be good to turn into a common sort function, as I can imagine us wanting to use it again.

No need to worry about it for this PR though, mostly calling it out for visibility/to remember to make it one day.